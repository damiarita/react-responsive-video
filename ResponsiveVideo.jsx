import React from 'react';

const ResponsiveVideo = () => {
      return (
        <picture/>
      );
    }
  
  export default ResponsiveVideo;