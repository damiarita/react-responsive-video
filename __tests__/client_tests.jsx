import React from 'react';
import { render, act } from '@testing-library/react';
import ResponsiveVideo from '../src'
import {pictureProps, imgProps, videoProps, sizes} from './data/fullProps'
import {sizes as minimalSizes} from './data/minProps'
import {sizes as invalidSizes} from './data/invalidProps'

let mockWindowWidth=0;
// Mock the useWindowWidth hook
jest.mock('@react-hook/window-size/throttled', () => ({
  useWindowWidth: jest.fn(()=>mockWindowWidth),
}));

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(function(query){
    const regex = /\(min-width: (\d*)px\)/;
    const match = regex.exec(query);
    const minWidth = parseInt(match[1]);
    return {
      matches: minWidth<=mockWindowWidth
  };}),
});

describe('Component on the browser', () => {

  test('renders with initial window width with full props', () => {
    // Mock initial window width
    mockWindowWidth=1024;

    const { asFragment } = render(<ResponsiveVideo
      pictureProps={pictureProps}
      imgProps={imgProps}
      videoProps={videoProps}
      sizes={sizes}
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with initial window width with minimal props', () => {
    // Mock initial window width
    mockWindowWidth=1024;

    const { asFragment } = render(<ResponsiveVideo
      sizes={minimalSizes}
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with initial window width with invalid props', () => {
    // Mock initial window width
    mockWindowWidth=1024;

    const { asFragment } = render(<ResponsiveVideo
      sizes={invalidSizes}
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('updates on window resize with full props', () => {
    // Mock initial window width
    mockWindowWidth=1024;
      const { asFragment, rerender } = render(<ResponsiveVideo
        pictureProps={pictureProps}
        imgProps={imgProps}
        videoProps={videoProps}
        sizes={sizes}
      />);
    expect(asFragment()).toMatchSnapshot();

    // Simulate a window resize event
    act(() => {
      mockWindowWidth=768;
      window.dispatchEvent(new Event('resize'));
    });
    rerender(<ResponsiveVideo
      pictureProps={pictureProps}
      imgProps={imgProps}
      videoProps={videoProps}
      sizes={sizes}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
