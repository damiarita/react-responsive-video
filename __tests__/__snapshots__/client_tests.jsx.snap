// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component on the browser renders with initial window width with full props 1`] = `
<DocumentFragment>
  <picture
    style="display: block;"
  >
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.avif"
      type="image/avif"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.webp"
      type="image/webp"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.jpg"
      type="image/jpeg"
      width="2000"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.avif"
      type="image/avif"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.webp"
      type="image/webp"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.jpg"
      type="image/jpeg"
      width="1024"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.avif"
      type="image/avif"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.webp"
      type="image/webp"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.jpg"
      type="image/jpeg"
      width="600"
    />
    <img
      id="testID"
    />
  </picture>
  <video
    height="1024"
    style="display: none;"
    width="1024"
  >
    <source
      src="https://www.example.com/tablet.mp4"
      type="video/mp4"
    />
    <source
      src="https://www.example.com/tablet.webm"
      type="video/webm"
    />
  </video>
</DocumentFragment>
`;

exports[`Component on the browser renders with initial window width with invalid props 1`] = `
<DocumentFragment>
  <picture>
    <source
      media="(min-width: 9999999px)"
      srcset="https://www.example.com/mobile.jpg"
      type="image/jpeg"
    />
    <img />
  </picture>
  <video
    style="display: none;"
  />
</DocumentFragment>
`;

exports[`Component on the browser renders with initial window width with minimal props 1`] = `
<DocumentFragment>
  <picture>
    <source
      srcset="https://www.example.com/mobile.jpg"
      type="image/jpeg"
    />
    <img />
  </picture>
  <video
    style="display: none;"
  >
    <source
      src="https://www.example.com/mobile.webm"
      type="video/webm"
    />
  </video>
</DocumentFragment>
`;

exports[`Component on the browser updates on window resize with full props 1`] = `
<DocumentFragment>
  <picture
    style="display: block;"
  >
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.avif"
      type="image/avif"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.webp"
      type="image/webp"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.jpg"
      type="image/jpeg"
      width="2000"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.avif"
      type="image/avif"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.webp"
      type="image/webp"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.jpg"
      type="image/jpeg"
      width="1024"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.avif"
      type="image/avif"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.webp"
      type="image/webp"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.jpg"
      type="image/jpeg"
      width="600"
    />
    <img
      id="testID"
    />
  </picture>
  <video
    height="1024"
    style="display: none;"
    width="1024"
  >
    <source
      src="https://www.example.com/tablet.mp4"
      type="video/mp4"
    />
    <source
      src="https://www.example.com/tablet.webm"
      type="video/webm"
    />
  </video>
</DocumentFragment>
`;

exports[`Component on the browser updates on window resize with full props 2`] = `
<DocumentFragment>
  <picture
    style="display: block;"
  >
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.avif"
      type="image/avif"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.webp"
      type="image/webp"
      width="2000"
    />
    <source
      height="972"
      media="(min-width: 1200px)"
      srcset="https://www.example.com/desktop.jpg"
      type="image/jpeg"
      width="2000"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.avif"
      type="image/avif"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.webp"
      type="image/webp"
      width="1024"
    />
    <source
      height="1024"
      media="(min-width: 768px)"
      srcset="https://www.example.com/tablet.jpg"
      type="image/jpeg"
      width="1024"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.avif"
      type="image/avif"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.webp"
      type="image/webp"
      width="600"
    />
    <source
      height="900"
      srcset="https://www.example.com/mobile.jpg"
      type="image/jpeg"
      width="600"
    />
    <img
      id="testID"
    />
  </picture>
  <video
    height="1024"
    style="display: none;"
    width="1024"
  >
    <source
      src="https://www.example.com/tablet.mp4"
      type="video/mp4"
    />
    <source
      src="https://www.example.com/tablet.webm"
      type="video/webm"
    />
  </video>
</DocumentFragment>
`;
